### ISSUE TITLE
Ways to reduce overfitting?

### ISSUE THREAD

**Issue by Sauravpandey98** [ultralytics/yolov5#7059]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello to all.I am working on project to detect warehouse boxes.So I trained a model with around 2000 images and tuned the parameters manually to get best model.After this I fine tuned my best model on a new dataset that consists of 1800 train images and around 484 validation images.Here here the stats of my dataset and training parameters:
# Dataset properties:
![media_images_Mosaics_0_1](https://user-images.githubusercontent.com/62154537/159168361-88e4230d-675d-42cc-b502-6005c327fd4d.jpg)
![media_images_Labels_0_0](https://user-images.githubusercontent.com/62154537/159168380-8b64259f-0131-45d5-b0b8-780add747522.jpg)

Results:
![Screenshot from 2022-03-20 20-31-07](https://user-images.githubusercontent.com/62154537/159168746-63b343ce-4bdc-4c97-b7e5-8fe37becbdac.png)
![Screenshot from 2022-03-20 20-30-49](https://user-images.githubusercontent.com/62154537/159168748-94c3868b-8dc7-45cc-909a-75e7690adf89.png)

So as you can see I have done most of the things as suggested by experts to reduce overfitting but still my model is overfitting very early.So I have a few questions:
1. As you can see the mAP values and precision and recall are good but still the obj loss seemed to be high even after reducing the hyp['obj].Why this is happening?
2. What can I do further to reduce overfitting?
3. Lastly if there is any problem with my fine tuning strategy then please tell me?

### Additional

And yes please ignore the other box labels that have no objects.It is just an error from my side(that has a long story :) ).It has no effect on training because all I want is to detect the first box label i.e 0th class.

**Comment by glenn-jocher** [comment]
@Sauravpandey98 see **Tips for Best Training Results** and **Hyperparameter Evolution** tutorials:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

Good luck 🍀 and let us know if you have any other questions!

**Comment by Sauravpandey98** [comment]
yes I have seen this and according to that I tried to apply augmentations and reduced hyp['object'] value to achieve desired result.But still it is overfitting . I cannot do hyper parameter evolution because of computation resources constrain.

**Comment by glenn-jocher** [comment]
@Sauravpandey98 wel, looking at your plots your mAP@0.5 is about 99%. Not sure what type of improvement you are trying to get but I'd say focus on increasing your dataset if you don't have resources for evolution.

**Comment by Sauravpandey98** [comment]
yes @glenn-jocher, sir yeah my mAP value is good.As you have said before that mAP value is affected by both regression loss and classification loss.So this mAP value may be because of very low  regression loss and a mediocre classification loss. I'm saying this because as you can see from the training graphs that even after reducing the hyp['obj'], the object loss is still not so low.Basically I want to reduce this loss further to make classification good.

So,my question is how can I reduce this loss. Also I may be wrong regarding this.So,correct me if I'm wrong.
